hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -105,7 +111,7 @@ public void doIn(@NonNull final TransferContext transferContext) {
                 }
 
                 final TokenAssociation newAssociation = validateAndBuildAutoAssociation(
-                        receiverId, tokenId, token, accountStore, tokenRelStore, handleContext);
+                        receiverId, tokenId, token, accountStore, tokenRelStore, handleContext, 0);
```

Comment:
Why is it 0 here ? We can do same check everytime right ?

Reply:
NftTransfer has no amount. Added amount comment in validateAndBuildAutoAssociation().